infrastructure/src/main/java/org/corfudb/infrastructure/LogReplicationServer.java

```diff
@@ -94,43 +81,51 @@ public void shutdown() {
 
     /* ************ Server Handlers ************ */
 
-    @ServerHandler(type = CorfuMsgType.LOG_REPLICATION_ENTRY)
-    private void handleLogReplicationEntry(CorfuPayloadMsg<LogReplicationEntry> msg, ChannelHandlerContext ctx, IServerRouter r) {
+    @RequestHandler(type = PayloadCase.LR_ENTRY_REQUEST)
+    private void handleLrEntryRequest(@Nonnull RequestMsg request,
+                                      @Nonnull ChannelHandlerContext ctx,
+                                      @Nonnull IServerRouter router) {
         log.trace("Log Replication Entry received by Server.");
 
-        if (isLeader(msg, r)) {
+        if (isLeader(request, ctx, router)) {
```

Comment:
`isLeader` has a pretty nasty side-effect -- is send back a response in case the node looses leadership. It is there to fulfill the interface requirements set by `IServerRouter`, which is implemented by `LogReplicationServerRouter`.

Reply:
This mostly has to do with the convention. When encountering functions of type `isSomething()` or `getSomething()`, the expectation is that such functions are side-effect free and they do not change the state of the system. Since this function sends back a leadership loss message, the above assertion does not hold.